hangups/client.py

```diff
@@ -149,6 +149,14 @@ def disconnect(self):
         When disconnection is complete, :func:`connect` will return.
         """
         logger.info('Disconnecting gracefully...')
+
+        # ignore a previous Exception
+        # pylint:disable=protected-access
+        if self._listen_future._exception is not None:
+            logger.info('disconnect: discard %s',
+                        repr(self._listen_future._exception))
+            self._client._listen_future._exception = None
```

Comment:
To avoid the protected access, could we instead call `self._listen_future.result()` and catch the exception?